libs/sdk-core/src/breez_services.rs

```diff
@@ -2073,6 +2071,48 @@ impl BreezServicesBuilder {
             persister: persister.clone(),
         });
 
+        // mempool space is used to monitor the chain
+        let mempoolspace_urls = match self.config.mempoolspace_url.clone() {
+            None => {
+                let cached = persister.get_mempoolspace_base_urls()?;
+                match cached.len() {
+                    0 => {
+                        // If we have no cached values, or we cached an empty list, fetch new ones
+
+                        let fresh_urls = breez_server
+                            .fetch_mempoolspace_urls()
+                            .await
+                            .unwrap_or(vec![DEFAULT_MEMPOOL_SPACE_URL.into()]);
+                        persister.set_mempoolspace_base_urls(fresh_urls.clone())?;
+                        fresh_urls
+                    }
+                    _ => {
+                        // If we already have cached values, return those
+
+                        // Start thread that refreshes cache in the background
+                        let cloned_breez_server = breez_server.clone();
+                        let cloned_persister = persister.clone();
+                        tokio::spawn(async move {
```

Comment:
Perhaps it is better to do that on start method? It is the point where currently we start all async tasks.

Reply:
I tried to, but unfortunately the `start` method only has access to `BreezServices`.

The new `fetch_mempoolspace_urls` method is on `BreezServer` itself. It's not exposed by any API (`NodeAPI`, `LspAPI`, etc) because it didn't seem to belong in any of them.

Should I expose it in an API, so I can call it in `start`? If so, which API makes more sense?

Reply:
@ok300  Perhaps we don't need to expose it in any of these traits and just create a BreezServer instance on demand for that? The same way we do with service_health_check?
We can add a new trait for settings but I feel it is overkill for now.

Reply:
Done: becd69ab685ab2eeef765b69ddf9cf2f82abcadb